apis/management.cattle.io/v3/rke_types.go

```diff
@@ -34,6 +34,29 @@ type PrivateRegistry struct {
 	Password string `yaml:"password" json:"password,omitempty"`
 }
 
+type RKESystemImages struct {
+	// etcd image
+	ETCDImage string `yaml:"etcd_image" json:"etcdImage,omitempty" norman:"default=quay.io/coreos/etcd:latest"`
+	// Alpine image
+	AlpineImage string `yaml:"alpine" json:"alpine,omitempty" norman:"default=alpine"`
+	// rke-nginx-proxy image
+	NginxProxyImage string `yaml:"nginx_proxy" json:"nginxProxy,omitempty" norman:"default=rancher/rke-nginx-proxy"`
+	// rke-cert-deployer image
+	CertDownloaderImage string `yaml:"cert_downloader" json:"certDownloader,omitempty" norman:"default=rancher/rke-cert-deployer"`
+	// rke-service-sidekick image
+	ServiceSidekickImage string `yaml:"service_sidekick_image" json:"serviceSideKickImage,omitempty" norman:"default=rancher/rke-service-sidekick"`
```

Comment:
Comment from @vincent99. Sidekick and sidecar mean the same; we should use consistent naming, up to you what to chose

Reply:
Sidecar is the name specified in the KubeDNS template, so we probably should keep that. 

For Sidekick, it's the name we use in 1.6 and it's currently the name of the yaml configuration tag and the container.. changing this will break backward compatibility.

Reply:
@moelsayed the image tag might not directly correlate to the parameter name on the api. On the api side of things we have to be consistent. Sidecar is the term widely used by k8s for rancher sidekick analog. So it is up to you what to chose - sidekick or sidecar - it just has to be consistent. 

Also I think at this point it is still fine to be not backwards compatible.